### PR TITLE
all: add LoadBalancer overload for Resolution results

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
This allows the load balancer to accept the service config object.   It is **not** yet wired up, this only creates the API do so.

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
This PR is not actually that big.  It's mostly churn in the unit tests.  The main files of interest are the LB implementations, and `LoadBalancer.java`

File: core/src/main/java/io/grpc/LoadBalancer.java [hunk 1]
```diff
@@ -121,11 +124,138 @@
...
+        List<EquivalentAddressGroup> servers,
+        @NameResolver.ResolutionResultAttr Attributes attributes,
+        Object loadBalancingPolicyConfig) {
+      this.servers =
+          Collections.unmodifiableList(new ArrayList<>(checkNotNull(servers, "servers")));
+      this.attributes = checkNotNull(attributes, "attributes");
+      this.loadBalancingPolicyConfig = loadBalancingPolicyConfig;
+    }
+
+    public static Builder newBuilder() {
```
[the remark this answers is not in the document]
Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: added.

File: core/src/main/java/io/grpc/LoadBalancer.java [hunk 2]
```diff
@@ -121,11 +124,138 @@
...
+      }
+
+      public Builder setLoadBalancingPolicyConfig(@Nullable Object loadBalancingPolicyConfig) {
+        this.loadBalancingPolicyConfig = loadBalancingPolicyConfig;
+        return this;
+      }
+
+      public ResolvedAddresses build() {
+        return new ResolvedAddresses(servers, attributes, loadBalancingPolicyConfig);
+      }
```
[the remark this answers is not in the document]
Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Done, PTAL.

[the remark this answers is not in the document]
Comment (zhangkun83) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Please document required/optional for each setter.

[the remark this answers is not in the document]
Comment (carl-mastrangelo) [reply core/src/main/java/io/grpc/LoadBalancer.java]: Done.

**Pull request merged**